server/src/main/java/org/opensearch/search/profile/query/InternalProfileComponent.java

```diff
@@ -31,6 +31,31 @@ public interface InternalProfileComponent {
      */
     long getTime();
 
+    /**
+     * @return the elapsed time in the reduce phase taken by this profile component
+     */
+    Long getReduceTime();
+
+    /**
+     * @return the maximum slice time taken by this profile component
+     */
+    Long getMaxSliceTime();
+
+    /**
+     * @return the minimum slice time taken by this profile component
+     */
+    Long getMinSliceTime();
+
+    /**
+     * @return the average slice time taken by this profile component
+     */
+    Long getAvgSliceTime();
+
+    /**
+     * @return the segment slice count taken by this profile component
+     */
+    int getSliceCount();
```

Comment:
These are specific to concurrent use case and will not be needed for non concurrent use case. For non concurrent use case the output is built by `InternalProfileCollector` whereas for concurrent case it will be built by `InternalProfileCollectorManager`. So all these methods and attributes being returned by these methods should be moved to `InternalProfileCollectorManager`

Reply:
Hi @sohami, thanks for your comment. Upon further consideration, your comment makes sense to me.